packages/react/src/products/hooks/useRecentlyViewedProducts.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  areRecentlyViewedProductsFetched,
+  areRecentlyViewedProductsLoading,
+  fetchRecentlyViewedProducts,
+  getRecentlyViewedProducts,
+  getRecentlyViewedProductsError,
+  getRecentlyViewedProductsPagination,
+  removeRecentlyViewedProduct,
+  saveRecentlyViewedProduct,
+} from '@farfetch/blackout-redux';
+import { useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction';
+import useProductListing from './useProductListing';
+import type { UseRecentlyViewedProductsOptions } from './types';
+
+const useRecentlyViewedProducts = (
+  productId: number,
```

Comment:
You can use either this:

```suggestion
  productId: ProductEntity['id'],
```
Or this:

```suggestion
  productId: Product['result']['id'],
```